File: Arrays/Medium/11/11.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        area = 0
        while l < r:
            dist = r - l
            temp = min(height[l], height[r])
            area = max(area, temp * dist)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return area

File: Arrays/Medium/11/11.c
int maxArea(int* height, int heightSize){
    int l = 0, r = heightSize - 1;
    int area = 0;
    while(l < r){
        int dist = r - l;
        int temp = height[l] < height[r] ? height[l] : height[r];
        area = area < temp * dist ? temp * dist : area;
        if(height[l] < height[r]){
            l++;
        }else{
            r--;
        }
    }
    return area;
}